Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1403,7 +1403,7 @@ PhysicalParticleContainer::PushP (int lev, Real dt,
 {
     WARPX_PROFILE("PhysicalParticleContainer::PushP");
 
-    if (do_not_push) return;
+    if (do_not_push || do_not_gather) return;
```

Comment:
Hm, just looking at the snippet here this looks quite confusing: why would the `do_not_gather` flag make the `PushP` step a no-OP... Can we control this at another place?

Reply:
I basically replicated what is already done for `PushPX` (see below). However, I agree with you that it is confusing. 
It would be more intuitive if `do_not_gather` just prevented the fields from the grid to be gathered (maybe considering just the external fields?) and only `do_not_push` were used to completely skip the push.
I could do that (both in `PushPx` and `PushP`)... 
```
void
PhysicalParticleContainer::PushPX (WarpXParIter& pti,
                                   amrex::FArrayBox const * exfab,
                                   amrex::FArrayBox const * eyfab,
                                   amrex::FArrayBox const * ezfab,
                                   amrex::FArrayBox const * bxfab,
                                   amrex::FArrayBox const * byfab,
                                   amrex::FArrayBox const * bzfab,
                                   const int ngE, const int /*e_is_nodal*/,
                                   const long offset,
                                   const long np_to_push,
                                   int lev, int gather_lev,
                                   amrex::Real dt, ScaleFields scaleFields,
                                   DtType a_dt_type)
{
    AMREX_ALWAYS_ASSERT_WITH_MESSAGE((gather_lev==(lev-1)) ||
                                     (gather_lev==(lev  )),
                                     "Gather buffers only work for lev-1");
    // If no particles, do not do anything
    // If do_not_gather = 1 by user, do not do anything
    if (np_to_push == 0 || do_not_gather) return;
```

Reply:
@RemiLehe and I looked a this and it would potentially introduce less side-effects if we place the `do_gather` guard as an `if` inside the `amrex::ParallelFor` to control the `doGatherShapeN( ... )` call. What do you think?

Reply:
I agree. It's what should be expected for a `do_not_gather` flag... I can do that.

Reply:
It should be OK now...

Reply:
Thanks a lot, looks great! Just a tiny change needed for a local variable and then it goes in :+1: